src/backend/joanie/tests/core/api/remote_endpoints/course_run/test_course_run.py

```diff
@@ -0,0 +1,573 @@
+            course=course,
+            state=models.CourseState.ONGOING_CLOSED,
+            languages="fr",
+        )
+        resource_link = f"{course_run.resource_link}".replace("+", "%2B")
+
+        response = self.client.get(
+            f"/api/v1.0/enrollments-and-orders-on-course-run/?resource_link={resource_link}",
+            HTTP_AUTHORIZATION="valid_known_secret_token_sample",
+        )
+
+        self.assertContains(
+            response,
+            "Make sure to give an existing resource link from an ended course run.",
+            status_code=400,
+        )
+
+    @override_settings(JOANIE_AUTHORIZED_API_TOKENS=["valid_known_secret_token_sample"])
+    def test_remote_endpoints_course_run_with_to_be_scheduled_course_run_state(
+        self,
+    ):
+        """
+        When the course run's end date is not yet reached, for example the state 'to be scheduled',
+        it should raise the error asking to provide a `resource_link` where the course run
+        has reached its end date.
+        """
+        course = factories.CourseFactory()
+        course_run = factories.CourseRunFactory(
+            is_listed=True,
+            course=course,
+            state=models.CourseState.TO_BE_SCHEDULED,
+            languages="fr",
+        )
+        resource_link = f"{course_run.resource_link}".replace("+", "%2B")
+
+        response = self.client.get(
+            f"/api/v1.0/enrollments-and-orders-on-course-run/?resource_link={resource_link}",
+            HTTP_AUTHORIZATION="valid_known_secret_token_sample",
+        )
+
+        self.assertContains(
+            response,
+            "Make sure to give an existing resource link from an ended course run.",
+            status_code=400,
+        )
+
+    @override_settings(JOANIE_AUTHORIZED_API_TOKENS=["valid_known_secret_token_sample"])
+    def test_remote_endpoints_course_run_authenticated_with_valid_token_and_the_required_parameter(
+        self,
+    ):
+        """
+        Authenticated user can query our remote endpoint if parsing valid token to the headers
+        with an existing `resource_link`. The token must be set into the settings variable
+        `JOANIE_AUTHORIZED_API_TOKENS`.
+        """
+        user = factories.UserFactory()
+        organization = factories.OrganizationFactory()
+        course = factories.CourseFactory(
+            organizations=[organization], users=[[user, enums.OWNER]]
+        )
+        course_run = factories.CourseRunFactory(
+            is_listed=True,
+            course=course,
+            state=models.CourseState.ARCHIVED_CLOSED,
+            languages="fr",
+        )
+        resource_link = f"{course_run.resource_link}".replace("+", "%2B")
+
+        response = self.client.get(
+            f"/api/v1.0/enrollments-and-orders-on-course-run/?resource_link={resource_link}",
+            HTTP_AUTHORIZATION="valid_known_secret_token_sample",
+        )
+
+        self.assertEqual(response.status_code, 200)
```

Comment:
Don't we want to check for `response.json()` here ?

Reply:
I will add it into the test